backend/tracim_backend/lib/core/content.py

```diff
@@ -1551,30 +1551,126 @@ def _copy_current(
+        self, content: Content, new_parent: Content = None
+    ) -> typing.Tuple[Content, typing.Dict[int, Content], typing.Dict[int, Content]]:
+        """
+        Create new content for content and his children, recreate all revision in order and
+        return all these new content
+        :param content: root content of copy
+        :param new_parent: new parent of root content
+        :return: new content created based on root content,
+        dict of new children content and original children content with original content id as key.
+        """
+        root_new_content = Content()
+        # INFO - G.M - 2019-04-30 - we store all children content created and old content id of them
+        # to be able to retrieve them for applying new revisions on them easily. key of dict is
+        # original content_id.
+        children_new_content = {}  # type: typing.Dict[int,Content]
+        # INFO - G.M - 2019-04-30 - we store alse old content of children to allow applying new
+        # revision related to old data. key of dict is original content id.
+        children_original_content = {}  # type: typing.Dict[int,Content]
+
+        for rev in content.recursive_tree_revision:
+            if rev.content_id == content.content_id:
+                related_content = root_new_content
+                related_parent = new_parent
+            else:
+                # INFO - G.M - 2019-04-30 - if we retrieve a revision without a new content related yet
+                # we create it.
+                if rev.content_id not in children_new_content:
+                    children_new_content[rev.content_id] = Content()
+                    children_original_content[rev.content_id] = rev.node
+                related_content = children_new_content[rev.content_id]  # type: Content
+                if rev.parent_id == content.content_id:
+                    related_parent = root_new_content
+                else:
+                    related_parent = children_new_content[rev.parent_id]
+            # INFO - G.M - 2019-04-30 - copy of revision itself.
+            cpy_rev = ContentRevisionRO.copy(rev, related_parent)
+            related_content.revisions.append(cpy_rev)
+            self._session.add(related_content)
+            self._session.flush()
+        return root_new_content, children_new_content, children_original_content
+
+    def _add_copy_revision(
+        self,
+        original_content: Content,
+        new_content: Content,
+        children_original_content,
+        children_new_content,
+        new_parent: Content = None,
+        new_label: str = None,
+        new_workspace: Workspace = None,
+        new_file_extension: str = None,
+        do_save: bool = True,
+        do_notify: bool = True,
+    ) -> typing.Tuple[Content, typing.Dict[int, Content], typing.Dict[int, Content]]:
+        """
+        Add copy revision for all new content
+        :param original_content: original content of root file in copy
```

Comment:
why "file" ? its' content, doesn't it? same remark for all parameter documented in the  docstring

Reply:
fixed.